internal/spokes/spokes.go

```diff
@@ -285,10 +300,9 @@ func (r *SpokesReceivePack) networkRepoPath() (string, error) {
 // isHiddenRef determines if the line passed as the first argument belongs to the list of
 // potential references that we don't want to advertise
 // This method assumes the config entries passed as a second argument are the ones in the `receive.hiderefs` section
-func isHiddenRef(line []byte, entries []config.ConfigEntry) bool {
-	l := string(line)
-	for _, entry := range entries {
-		if strings.Contains(l, entry.Value) {
+func isHiddenRef(line string, hiddenRefs []string) bool {
+	for _, hr := range hiddenRefs {
+		if strings.Contains(line, hr) {
```

Comment:
Should this use `strings.HasPrefix` instead of `strings.Contains`? The docs say that `transfer.hiderefs` and `receive.hiderefs` are prefixes.

Should this implement handling for `!` or `^` modifiers? In particular, babeld uses the negation `!` operator during [internal pushes from svnbridge](https://github.com/github/babeld/blob/89fc82eff6e064dedf23206811c18ea074909ef8/src/util.c#L866)).

Reply:
I have modified it here b360a290d1e3dfdc8da0f59683e5e42b297de71d

I am not sure if this would need to implement the handling of such elements, but if Babeld sends them we will probably need to support them? This hasn't been implemented in the previous commit

Reply:
I am pretty sure we'll need to support `!` at least. I'm not sure what the right ordering would be.